pkg/trace/sampler/catalog.go

```diff
@@ -83,8 +79,7 @@ func (cat *serviceKeyCatalog) ratesByService(agentEnv string, rates map[Signatur
 			delete(cat.items, key)
 			continue
 		}
-
-		if rateWithEmptyEnv(key.Env, agentEnv) {
+		if key.Env == "" {
```

Comment:
I don't know why the previous `rateWithEmptyEnv` did this:
```
return samplerEnv == agentEnv
```
instead of just checking if `key.Env == ""`, but I think this should be safer?

Reply:
Oh yep you're right. When I deleted it earlier, some tests were failing. But they are passing now, and I don't recall what the exact issues were.